src/shared/watchdog.c

```diff
@@ -115,8 +118,18 @@ int watchdog_set_timeout(usec_t *usec) {
 }
 
 int watchdog_ping(void) {
+        usec_t ntime;
         int r;
 
+        ntime = now(clock_boottime_or_monotonic());
+        assert(ntime >= watchdog_last_ping);
```

Comment:
i just noticed that we'll call watchdog_ping() even if the watchdog isn't open, in which case watchdog_last_ping will still be USEC_INFINITY, i.e. this will hit the assert. maybe condition this whole addition out if watchdog_last_ping() is not initialized yet.

Reply:
i.e. move this into the second if branch below

Reply:
I also had some concerns about that assert. I will check again possible executions scenarios and do the requested updates. I'll try also to do the rebase and squash the commits. I'm not that familiar with GitHub workflow so I have to check documentation a little bit.

Reply:
OK, I squashed the commits and fix the assert issue, I'll check the CI results later today.